src/exo/pyparser.py

```diff
@@ -166,15 +172,18 @@ def __init__(
         self.is_fragment = is_fragment
 
         self.push()
+        special_cases = ["stride"]
```

Comment:
Creating this special_cases list and then searching it linearly (in the `s.value.func.id in special_cases`) looks like potentially expensive "accumulated cost" considering how many times this parser is used, e.g., to parse all procs and all  patterns in `@instr`. Can this be an `is_special_case` lambda that just searches for the func id in the 2 namespaces (+ the special stride case)?

Reply:
Something like

```
        def is_special_case(func_id):
            return (
                func_id == "stride"
                or isinstance(self.globals.get(func_id), Extern)
                or isinstance(self.locals.get(func_id), Extern)
            )
...
            if len(module_ast) == 1:
                s = module_ast[0]
                if isinstance(s, pyast.Expr) and (
                    not isinstance(s.value, pyast.Call)
                    or is_special_case(s.value.func.id)
                ):
                    is_expr = True
```

Reply:
i think `special_cases` is only used once which is to check patterns that are of the form `<extern>(...)` so I don't think there is a big cost to doing this

Reply:
I think this is fine